instance/instance.go

```diff
@@ -610,28 +623,70 @@ func writeBody(w http.ResponseWriter, b []byte) {
 // PublishCheck Checks if an instance has been published
 type PublishCheck struct {
 	Datastore store.Storer
+	Auditor   audit.AuditorService
 }
 
 // Check wraps a HTTP handle. Checks that the state is not published
-func (d *PublishCheck) Check(handle func(http.ResponseWriter, *http.Request)) http.HandlerFunc {
+func (d *PublishCheck) Check(handle func(http.ResponseWriter, *http.Request), action string) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-
+		ctx := r.Context()
 		vars := mux.Vars(r)
-		id := vars["id"]
-		instance, err := d.Datastore.GetInstance(id)
-		if err != nil {
-			log.Error(err, nil)
-			handleErrorType(err, w)
+		instanceID := vars["id"]
+		logData := log.Data{"instance_id": instanceID}
+		auditParams := common.Params{"instance_id": instanceID}
+
+		if err := d.Auditor.Record(ctx, action, audit.Attempted, auditParams); err != nil {
+			audit.LogActionFailure(ctx, action, audit.Attempted, err, logData)
+			handleErr(ctx, errs.ErrAuditActionAttemptedFailure, 0, w, logData)
 			return
 		}
 
-		if instance.State == models.PublishedState {
-			err = errors.New("unable to update instance as it has been published")
-			log.Error(err, log.Data{"instance": instance})
-			http.Error(w, err.Error(), http.StatusForbidden)
+		statusCode, err := d.checkState(w, instanceID)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "errored whilst checking instance state"), logData)
+			if auditErr := d.Auditor.Record(ctx, action, audit.Unsuccessful, auditParams); auditErr != nil {
+				audit.LogActionFailure(ctx, action, audit.Unsuccessful, auditErr, logData)
+				handleErr(ctx, errs.ErrAuditActionAttemptedFailure, 0, w, logData)
```

Comment:
The error message for `errs.ErrAuditActionAttemptedFailure` is internal server error. Do we want something more specific to suggest it was auditing that failed? I am guessing this message was set to satisfy existing tests? Not too big a deal if the logs are descriptive, just seemed a bit odd

Reply:
I don't believe we do want to tell api customers the reason for an internal server error, the logs will capture that it was due to an auditing failure.

Reply:
Yea I figured that was the case. I think the bit that feels odd is that the error messages end up in the http responses. I guess we should map an error type to a http response message in the API, and have the error message specific to the error. (not one for this PR, as this is the case in most places)

Reply:
Very true, I think I will fix this here and inform the team in standup